### ISSUE TITLE
Move from AppVeyor to GitHub Actions

### ISSUE THREAD

**Issue by jonorossi** [castleproject/Core#497]
I'm using GitHub Actions for a couple of private projects and have been happy. The integration into the GitHub UI is nicer, we'd be able to have more concurrent jobs which does get annoying with AppVeyor being limited to just one. I haven't tried out GitHub Packages, but I'd be interested in seeing if that would work well for CI builds.

Today we build and test on .NET Framework and .NET Core on Windows, and Mono and .NET Core on Linux, however it would be good to build our .NET Standard package once, and test it across all 3 platforms. GitHub Actions jobs can store files for another job for matrix builds.

Thoughts?

**Comment by stakx** [comment]
While fixing our AppVeyor CI build, I was about to suggest moving to GitHub Actions, too. I ended up not to, because in the short term, I'd rather invest my time in getting the next 1-2 versions of Castle.Core ready, rather than rewriting the CI build. 😄

Once we have Castle.Core 5 done and published, I think moving to GitHub Actions would be a good idea. When we moved our Linux build from Travis to Appveyor, I thought it would be nice that Appveyor automatically manages & updates Mono for us. As the current breakage shows, that isn't so nice after all, and having a little bit more control over build-time dependencies would be good. GitHub Actions might give us that. (Alternatively, we could just reduce our use of AppVeyor as a Docker trampoline.)

TL;DR: I'm all for GitHub Actions, but would be glad if we could get Castle.Core v5 released first.

**Comment by jonorossi** [comment]
[Dustin Moris Gorski's blog post](https://dusted.codes/github-actions-for-dotnet-core-nuget-packages) yesterday might be useful, he even has the Giraffe project builds publishing to GitHub Packages.
